aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -1377,7 +1413,27 @@ AT_ERROR("svd: MAGMA library not found in "
 
   magma_int_t m = magma_int_cast(self.size(-2), "m");
   magma_int_t n = magma_int_cast(self.size(-1), "n");
-  auto k = std::min(m, n);
+  auto mn = std::min(m, n);
+
+  c10::Storage storage_rwork;
+  value_t* rwork = nullptr;
+
+  magma_int_t* iwork;
+  ALLOCATE_ARRAY(iwork, magma_int_t, 8 * mn);
+  // Copy-n-paste rwork size computation from BatchLinearAlgebra.cpp
+  if (isComplexType(at::typeMetaToScalarType(self.dtype()))) {
+    auto mx = std::max(m, n);
+    int64_t lrwork; // These settings are valid for on LAPACK 3.6+
+    if (jobz == MagmaNoVec){
+      lrwork = 7 * mn;
+    } else if (mx > 10 * mn){
+      lrwork = 7 * mn * mn + 7 * mn;
+    } else {
+      lrwork = std::max(7 * mn * mn + 7 * mn, 2 * mx * mn + 2 *mn * mn + mn);
+    }
```

Comment:
Slightly less memory will be reserved.
```suggestion
    int64_t lrwork; // These settings are valid for MAGMA and LAPACK 3.7+
    if (jobz == MagmaNoVec){
      lrwork = 5 * mn;
    } else if (mx > 10 * mn){
      lrwork = 5 * mn * mn + 5 * mn;
    } else {
      lrwork = std::max(5 * mn * mn + 5 * mn, 2 * mx * mn + 2 *mn * mn + mn);
    }
```

Reply:
Thank you for the feedback.
As mentioned in the comment above this block, it's a copy-n-paste from BatchLinearAlgebra.cpp
Once this change lands, I will create a separate PR that moves this logic to LinearAlgebraUtils and optimizes memory utilization.

Reply:
See https://github.com/pytorch/pytorch/pull/45812